scripts/annotation_tool.py

```diff
@@ -0,0 +1,787 @@
+        center = self._annotation_scene.annotation_scene.get_center()
+        center[2] -= 0.2
+        init_trans[0:3, 3] = center
+        object_geometry.transform(init_trans)
+        new_mesh_instance = self._obj_instance_count(self._meshes_available.selected_value, meshes)
+        new_mesh_name = str(self._meshes_available.selected_value) + '_' + str(new_mesh_instance)
+        self._scene.scene.add_geometry(new_mesh_name, object_geometry, self.settings.annotation_obj_material,
+                                       add_downsampled_copy_for_fast_rendering=True)
+        self._annotation_scene.add_obj(object_geometry, new_mesh_name, new_mesh_instance, transform=init_trans)
+        meshes = self._annotation_scene.get_objects()  # update list after adding current object
+        meshes = [i.obj_name for i in meshes]
+        self._meshes_used.set_items(meshes)
+        self._meshes_used.selected_index = len(meshes) - 1
+
+    def _remove_mesh(self):
+        if not self._annotation_scene.get_objects():
+            print("There are no object to be deleted.")
+            return
+        meshes = self._annotation_scene.get_objects()
+        active_obj = meshes[self._meshes_used.selected_index]
+        self._scene.scene.remove_geometry(active_obj.obj_name)  # remove mesh from scene
+        self._annotation_scene.remove_obj(self._meshes_used.selected_index)  # remove mesh from class list
+        # update list after adding removing object
+        meshes = self._annotation_scene.get_objects()  # get new list after deletion
+        meshes = [i.obj_name for i in meshes]
+        self._meshes_used.set_items(meshes)
+
+    def _make_point_cloud(self, rgb_img, depth_img, cam_K):
+        # convert images to open3d types
+        rgb_img_o3d = o3d.geometry.Image(cv2.cvtColor(rgb_img, cv2.COLOR_BGR2RGB))
+        depth_img_o3d = o3d.geometry.Image(depth_img)
+
+        # convert image to point cloud
+        intrinsic = o3d.camera.PinholeCameraIntrinsic(rgb_img.shape[0], rgb_img.shape[1],
+                                                      cam_K[0, 0], cam_K[1, 1], cam_K[0, 2], cam_K[1, 2])
+        rgbd = o3d.geometry.RGBDImage.create_from_color_and_depth(rgb_img_o3d, depth_img_o3d,
+                                                                  depth_scale=1, convert_rgb_to_intensity=False)
+        pcd = o3d.geometry.PointCloud.create_from_rgbd_image(rgbd, intrinsic)
+
+        return pcd
+
+    def scene_load(self, scenes_path, scene_num, image_num):
+        self._annotation_changed = False
+
+        self._scene.scene.clear_geometry()
+        geometry = None
+
+        scene_path = os.path.join(scenes_path, f'{scene_num:06}')
+        rgb_path = os.path.join(scene_path, 'rgb', f'{image_num:06}' + '.png')
+        rgb_img = cv2.imread(rgb_path)
+        depth_path = os.path.join(scene_path, 'depth', f'{image_num:06}' + '.png')
+        depth_img = cv2.imread(depth_path, -1)
+        depth_img = np.float32(depth_img / 1000)
```

Comment:
This assumes that the loaded input depth maps are always in `mm`. In BOP datasets we save depth images in 16bit and to increase resolution we use a `depth_scale` parameter that scales the depth to mm. 

It is saved in the `scene_camera.json` you load below. You can also set a default value of 1.0 if it does not exist.

So you would do something like:
`depth_img = np.float32(depth_img * depth_scale / 1000)`

Reply:
added